File: retroauth-android/src/test/java/com/andretietz/retroauth/WeakActivityStackTest.java
package com.andretietz.retroauth;

import android.app.Activity;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;

@RunWith(RobolectricTestRunner.class)
@Config(constants = BuildConfig.class, sdk = 23)
public class WeakActivityStackTest {


    WeakActivityStack stack;

    @Before
    public void setup() {
        stack = new WeakActivityStack();
    }

    @Test
    public void basicStackTest() {
        Activity activity1 = mock(Activity.class);
        Activity activity2 = mock(Activity.class);
        Activity activity3 = mock(Activity.class);

        stack.push(activity1);
        stack.push(activity2);
        stack.push(activity3);

        assertEquals(activity3, stack.pop());
        assertEquals(activity2, stack.pop());
        assertEquals(activity1, stack.pop());
    }


    @Test
    public void stackRemovalTest() {
        Activity activity1 = mock(Activity.class);
        Activity activity2 = mock(Activity.class);
        Activity activity3 = mock(Activity.class);

        stack.push(activity1);
        stack.push(activity2);
        stack.push(activity3);

        stack.remove(activity2);

        assertEquals(activity3, stack.pop());
        assertEquals(activity1, stack.pop());
    }

    @Test
    public void splashScreenTest() {
        Activity splashScreen = mock(Activity.class);
        Activity mainScreen = mock(Activity.class);

        stack.push(splashScreen);
        stack.push(mainScreen);

        stack.remove(splashScreen);

        assertEquals(mainScreen, stack.peek());
    }


}

File: android-accountmanager/src/test/java/com/andretietz/retroauth/WeakActivityStackTest.kt
package com.andretietz.retroauth

import android.app.Activity
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito.mock

@RunWith(AndroidJUnit4::class)
class WeakActivityStackTest {

  private val stack = WeakActivityStack()

  @Test
  fun basicStackTest() {
    val activity1 = mock(Activity::class.java)
    val activity2 = mock(Activity::class.java)
    val activity3 = mock(Activity::class.java)

    stack.push(activity1)
    stack.push(activity2)
    stack.push(activity3)

    assertEquals(activity3, stack.pop())
    assertEquals(activity2, stack.pop())
    assertEquals(activity1, stack.pop())
  }

  @Test
  fun stackRemovalTest() {
    val activity1 = mock(Activity::class.java)
    val activity2 = mock(Activity::class.java)
    val activity3 = mock(Activity::class.java)

    stack.push(activity1)
    stack.push(activity2)
    stack.push(activity3)

    stack.remove(activity2)

    assertEquals(activity3, stack.pop())
    assertEquals(activity1, stack.pop())
  }

  @Test
  fun splashScreenTest() {
    val splashScreen = mock(Activity::class.java)
    val mainScreen = mock(Activity::class.java)

    stack.push(splashScreen)
    stack.push(mainScreen)

    stack.remove(splashScreen)

    assertEquals(mainScreen, stack.peek())
  }
}
